crates/core/src/network/actions/shutdown.rs

```diff
@@ -30,12 +38,18 @@ pub async fn shutdown(
 
 pub struct ShutdownFuture {
     state: Arc<RwLock<StateWrapper>>,
+    running_time: Instant,
 }
 
 impl Future for ShutdownFuture {
     type Output = HcResult<()>;
 
     fn poll(self: Pin<&mut Self>, cx: &mut std::task::Context) -> Poll<Self::Output> {
+        if self.running_time.elapsed() > Duration::from_secs(70) {
+            panic!("future has been running for too long")
```

Comment:
This is a different style of future diagnostics. I would suggest we harmonize and have at least all be the same. This one might be less performance problematic..